chain/network/src/network_protocol/mod.rs

```diff
@@ -90,6 +90,36 @@ impl std::str::FromStr for PeerAddr {
     }
 }
 
+/// Proof that a given peer_id owns an ip address, included in Handshake message
```

Comment:
Can we revise this comment to say what's really going on here? As mentioned previously ip addresses aren't really "owned." I would suggest something like:
```
// Signed ip address included as part of the Handshake message.
// Used to authenticate identity of connected peers.
// Necessary because peer discovery happens via unauthenticated PeerInfos.
```